x/ibc/03-connection/keeper/handshake.go

```diff
@@ -164,47 +152,31 @@ func (k Keeper) ConnOpenAck(
 		)
 	}
 
-	// XXX: blocked by #5078
-	/*
-		expectedConsensusState, found := k.clientKeeper.GetConsensusState(ctx, connection.ClientID)
-		if !found {
-			return errors.New("client consensus state not found") // TODO: use ICS02 error
-		}
-	*/
+	expectedConsensusState, found := k.clientKeeper.GetConsensusState(ctx, connection.ClientID, consensusHeight)
+	if !found {
+		return clienttypes.ErrConsensusStateNotFound
+	}
+
 	prefix := k.GetCommitmentPrefix()
 	expectedCounterparty := types.NewCounterparty(connection.ClientID, connectionID, prefix)
-	expectedConn := types.NewConnectionEnd(types.TRYOPEN, connection.Counterparty.ClientID, expectedCounterparty, []string{version})
+	expectedConnection := types.NewConnectionEnd(exported.TRYOPEN, connection.Counterparty.ClientID, expectedCounterparty, []string{version})
 
-	expConnBz, err := k.cdc.MarshalBinaryLengthPrefixed(expectedConn)
-	if err != nil {
+	if err := k.VerifyConnectionState(
+		ctx, proofHeight, proofTry, connection.Counterparty.ConnectionID,
+		expectedConnection, expectedConsensusState,
+	); err != nil {
 		return err
 	}
 
-	ok := k.VerifyMembership(
-		ctx, connection, proofHeight, proofTry,
-		types.ConnectionPath(connection.Counterparty.ConnectionID), expConnBz,
-	)
-	if !ok {
-		return errors.New("couldn't verify connection membership on counterparty's client") // TODO: sdk.Error
-	}
-
 	// XXX: blocked by #5078
-	/*
-		expConsStateBz, err := k.cdc.MarshalBinaryLengthPrefixed(expectedConsensusState)
-		if err != nil {
-			return err
-		}
-
-		ok = k.VerifyMembership(
-			ctx, connection, proofHeight, proofConsensus,
-			clienttypes.ConsensusStatePath(connection.Counterparty.ClientID), expConsStateBz,
-		)
-		if !ok {
-			return errors.New("couldn't verify consensus state membership on counterparty's client") // TODO: sdk.Error
-		}
-
-	*/
-	connection.State = types.OPEN
+	// err = k.VerifyClientConsensusState(
```

Comment:
What is this blocked by? https://github.com/cosmos/cosmos-sdk/issues/5078 is closed.